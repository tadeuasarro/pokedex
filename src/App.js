import React from 'react';
import './App.css';

const app = () => (
  <div>
    something amazing
  </div>
)

export default App;
