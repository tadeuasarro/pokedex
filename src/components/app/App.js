import React from 'react';
import KantoList from '../../containers/kantolist/KantoList';
import Filtered from '../../containers/filtered/Filtered';
import Navbar from '../navbar/Navbar';
import './app.css';

const App = () => (
  <div>
    <Navbar />
    <Filtered />
    <KantoList />
  </div>
);

export default App;
