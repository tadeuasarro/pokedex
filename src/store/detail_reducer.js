import { FETCH_DETAIL_ERROR, FETCH_DETAIL_PENDING, FETCH_DETAIL_SUCCESS } from '../actions/index';

const detailInitial = {
  pending: false,
  results: false,
  error: null,
};

export const detailReducer = (state = detailInitial, action) => {
  switch (action.type) {
    case FETCH_DETAIL_PENDING:
      return {
        ...state,
        pending: true,
      };
    case FETCH_DETAIL_SUCCESS:
      return {
        ...state,
        pending: false,
        results: action.detail,
      };
    case FETCH_DETAIL_ERROR:
      return {
        ...state,
        pending: false,
        error: action.error,
      };
    default:
      return state;
  }
};

export const getDetail = state => state.detail;
export const getDetailPending = state => state.pending;
export const getDetailError = state => state.error;
