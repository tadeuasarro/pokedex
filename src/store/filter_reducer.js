import { FETCH_TYPES_ERROR, FETCH_TYPES_PENDING, FETCH_TYPES_SUCCESS } from '../actions/index';

const detailInitial = {
  pending: false,
  results: false,
  error: null,
};

export const filterReducer = (state = detailInitial, action) => {
  switch (action.type) {
    case FETCH_TYPES_PENDING:
      return {
        ...state,
        pending: true,
      };
    case FETCH_TYPES_SUCCESS:
      return {
        ...state,
        pending: false,
        results: action.types,
      };
    case FETCH_TYPES_ERROR:
      return {
        ...state,
        pending: false,
        error: action.error,
      };
    default:
      return state;
  }
};

export const getTypes = state => state.types;
export const getTypesPending = state => state.pending;
export const getTypesError = state => state.error;
