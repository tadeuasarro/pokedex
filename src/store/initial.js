const pokemonInitial = {
  pending: false,
  results: [],
  error: null,
  detail: false,
};

export default pokemonInitial;
