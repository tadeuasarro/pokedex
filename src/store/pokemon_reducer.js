import { FETCH_POKEMON_PENDING, FETCH_POKEMON_SUCCESS, FETCH_POKEMON_ERROR } from '../actions/index';

const pokemonInitial = {
  pending: false,
  results: [],
  error: null,
};

export const pokemonReducer = (state = pokemonInitial, action) => {
  switch (action.type) {
    case FETCH_POKEMON_PENDING:
      return {
        ...state,
        pending: true,
      };
    case FETCH_POKEMON_SUCCESS:
      return {
        ...state,
        pokemon: action.pokemon,
      };
    case FETCH_POKEMON_ERROR:
      return {
        ...state,
        pending: false,
        error: action.error,
      };
    default:
      return state;
  }
};

export const getPokemon = state => state.results;
export const getPokemonPending = state => state.pending;
export const getPokemonError = state => state.error;
