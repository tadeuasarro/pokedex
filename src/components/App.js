import React from 'react';
import './App.css';

const App = () => (
  <div>
    something amazing
  </div>
);

export default App;
