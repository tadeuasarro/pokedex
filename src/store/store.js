import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { pokemonReducer } from './pokemon_reducer';
import { detailReducer } from './detail_reducer';

const store = createStore(
  combineReducers({
    pokemon: pokemonReducer,
    detail: detailReducer,
  }),
  applyMiddleware(thunk),
);

export default store;
