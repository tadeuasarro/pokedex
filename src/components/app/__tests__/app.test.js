import { render, screen, act } from '@testing-library/react';
import renderer from 'react-test-renderer';
import store from '../../../store/store';
import { Provider } from 'react-redux';
import App from '../App';

describe('Rendering component', () => {
  it('creates an App component', () => {
    act(() => {
      render(<Provider store={store}><App /></Provider>);
    })
    const element = screen.getByText('Filter');
  })
})

describe('Display', () => {
  it('renders correctly', () => {
    const comp = renderer.create(<Provider store={store}><App /></Provider>).toJSON();
    expect(comp).toMatchSnapshot();
  })
})

