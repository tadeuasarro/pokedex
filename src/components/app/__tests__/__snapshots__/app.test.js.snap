// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display renders correctly 1`] = `
<div>
  <div
    id="navbar"
  >
    <img
      alt="PokÃ©mon"
      className="pokemon-logo"
      src="./logo.png"
    />
  </div>
  <div
    className="filter-container"
  >
    <form>
      <select
        id="types-listing"
      >
        <option
          value="all"
        >
          All pokemon
        </option>
        <option
          value="bug"
        >
          Bug
        </option>
        <option
          value="dragon"
        >
          Dragon
        </option>
        <option
          value="electric"
        >
          Electric
        </option>
        <option
          value="fighting"
        >
          Fighting
        </option>
        <option
          value="fire"
        >
          Fire
        </option>
        <option
          value="flying"
        >
          Flying
        </option>
        <option
          value="ghost"
        >
          Ghost
        </option>
        <option
          value="grass"
        >
          Grass
        </option>
        <option
          value="ground"
        >
          Ground
        </option>
        <option
          value="ice"
        >
          Ice
        </option>
        <option
          value="normal"
        >
          Normal
        </option>
        <option
          value="poison"
        >
          Poison
        </option>
        <option
          value="psychic"
        >
          Psychic
        </option>
        <option
          value="rock"
        >
          Rock
        </option>
        <option
          value="water"
        >
          Water
        </option>
      </select>
      <button
        className="filter-button"
        onClick={[Function]}
        type="button"
      >
        Filter
      </button>
    </form>
  </div>
  <div
    className="loading-container"
  >
    <h1
      className="loading-message"
    >
      Loading...
    </h1>
  </div>
</div>
`;
